plugin/src/server/sc/plugin2021/GamePlugin.kt

```diff
@@ -28,15 +31,16 @@ class GamePlugin: IGamePlugin {
                     Color::class.java, Team::class.java)
         
         fun registerXStream() {
-            LobbyProtocol.registerAdditionalMessages(xstream, classesToRegister)
-            
-            xstream.registerConverter(BoardConverter())
+            if (loaded.compareAndSet(false, true)) {
```

Comment:
if this atomic compare and set is necessary, you'd also want to lock threads during execution of the body.
Otherwise...
I think execution of this body twice is better than xstream access without prior loading, in which case you'd want to set the flag at the end of this block. I'm not sure how xstream handles duplicate registers, though